src/uu/uniq/src/uniq.rs

```diff
@@ -247,6 +250,40 @@ fn opt_parsed<T: FromStr>(opt_name: &str, matches: &ArgMatches) -> UResult<Optio
     })
 }
 
+// gets number of fields to be skipped from the shorthand option `-N`
+//
+//```
+//uniq -12345
+//```
+// the first digit isn't interpreted by clap as part of the value
+// so `get_one()` would return `2345`, then to get the actual value
+// we loop over every possible first digit, append the value to it
+// and parse the resulting string as usize, an error at this point
+// means that a character which isn't a digit was given
+fn obsolete_skip_field(matches: &ArgMatches) -> UResult<Option<usize>> {
+    for opt_text in OBSOLETE_SKIP_FIELDS_DIGITS {
+        let argument = matches.get_one::<String>(opt_text);
+        if matches.contains_id(opt_text) {
+            let mut full = opt_text.to_owned();
+            if let Some(ar) = argument {
+                full.push_str(ar);
+            }
+            let value = full.parse::<usize>();
+
+            if let Err(_reason) = value {
+                return Err(USimpleError {
+                    code: 42,
```

Comment:
Where does 42 come from? `uniq` doesn't fail in this case, but `fold` exits with 1. We should probably follow that.

Also, if you chose 42 just as a random value, please ask what it should be next time and don't make it up, because I almost missed this while reviewing.

Reply:
crap, i forgot that, i put in a different number because i wanted to see  if it was the return value, thanks